e2e/test/Helpers/TestDevice.cs

```diff
@@ -170,5 +176,25 @@ public async Task RemoveDeviceAsync()
             using var rm = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
             await rm.RemoveDeviceAsync(Id).ConfigureAwait(false);
         }
+
+        private static async Task RetryOnThrottlingAsync(Func<Task> asyncOperation)
+        {
+            int counter = 0;
+            do
+            {
+                try
+                {
+                    await asyncOperation().ConfigureAwait(false);
```

Comment:
Do you know if there is information in the response on when to retry? Usually all services should send that value back to us in the response. Our tests would be less flaky if we use that. If that information is not there, consider using an exponential back-off. The chances of success is low if we have a linear back-off.

Reply:
Unfortunately service doesn't give us a retry-after value, but rather indicates in the message directly that we should retry after 10 secs: Microsoft.Azure.Devices.Common.Exceptions.ThrottlingException: {"Message":"ErrorCode:ThrottlingBacklogTimeout;The request has been throttled. Wait 10 seconds and try again. Operation type: CRUD","ExceptionMessage":"Tracking ID:04c922b3aac44715a7569eef9e513463-G:4-TimeStamp:03/23/2021 17:00:30"}

Good point about introducing an exponential backoff, I'll add that in.